styx-api-service/src/test/java/com/spotify/styx/api/BackfillResourceTest.java

```diff
@@ -342,6 +352,32 @@ public void shouldGetBackfillStatusWithStatusByDefault() throws Exception {
     assertJson(response, "statuses.active_states", hasSize(24));
   }
 
+  @Test
+  public void shouldGetBackfillStatusWithStatusByDefaultReversed() throws Exception {
+    sinceVersion(Api.Version.V3);
+
+    WorkflowInstance wfi = WorkflowInstance.create(BACKFILL_2.workflowId(), "2017-01-01T23");
```

Comment:
storage should be mocked out in these tests

Reply:
Maybe there's a bit too much event replay logic in `BackfillResource` itself. Factoring out that should ease testing. I.e., it should be possible to mock `getBackfillRunState()` in `BackfillResourceTest` and test `getBackfillRunState()` in a separate `ReplayEventsTest`.

Reply:
can be another PR I guess